### PR TITLE
Display full websocket message

### PR DESCRIPTION
**Pull request by psiinon**
This is part 2 of #373 it displays the message but I havnt implemented resend yet.

### CODE REVIEW

**Comment by psiinon** [comment]
Just noticed that incoming messages are still flagged as 'Outgoin' - will need to fix that.
I'll also change the table to use the 1i8n'd direction strings too.

**Comment by psiinon** [comment]
Fixed those problems. Also fixed the problem where JSON payloads were being displayed as [Object]

**Comment by thc202** [comment]
The side effect of `JSON.stringify` the payload is that it might change it, special chars are quoted and the content itself, if plain string.

**Comment by psiinon** [comment]
This is a bug in the websockets add-on - I'll put a PR in to fix that.

**Comment by psiinon** [comment]
Have fixed the JSON encoding bugs in https://github.com/zaproxy/zap-extensions/pull/1954
so undone the conversion thats no longer needed here.
Hopefully good to go now...

File: src/main/zapHomeFiles/hud/display.html [hunk 1]
```diff
@@ -278,6 +279,37 @@
...
+                            <tr>
+                                <td> {{ $t('message.websockets_message_field_payload') }} </td>
+                                <td> </td>
+                            </tr>
+                        </tbody>
+                    </table>
+                    <textarea class="form-input hud-websocket-message" rows="10" spellcheck="false" v-model="payload" :readonly="true"></textarea>
+                </div>
+                <div slot="footer">
+                    <!-- button class="btn btn-primary" @click="replay"> {{ $t('message.websockets_replay') }} </button-->
```
Comment (Pamplemousse) [inline src/main/zapHomeFiles/hud/display.html]: Comment leftover.

File: src/main/zapHomeFiles/hud/display.js [hunk 2]
```diff
@@ -438,6 +438,49 @@ Vue.component('history-message-modal', {
...
 
+Vue.component('websocket-message-modal', {
+	template: '#websocket-message-modal-template',
+	props: ['show', 'title'],
+	methods: {
+		close: function() {
+			this.$emit('close');
+		},
+		replay: function() {
+			// TODO implement
```
Comment (Pamplemousse) [inline src/main/zapHomeFiles/hud/display.js]: `TODO` leftover.

File: src/main/zapHomeFiles/hud/tools/websockets.js [hunk 3]
```diff
@@ -45,6 +45,28 @@ var WebSockets = (function() {
...
+		if (!data) {
+			throw new Error('Could not load WebSocket message details')
+		}
+
+		return utils.messageFrame(tabId, "display", {action:"showWebSocketMessage", config:data})
+		.then(data => {
+			// Handle button choice
+			if (data.buttonSelected === "replay") {
+				utils.log (LOG_DEBUG, 'websockets.showWebSocketMessageDetails', 'replay', data);
+				// TODO implement
```
Comment (Pamplemousse) [inline src/main/zapHomeFiles/hud/tools/websockets.js]: `TODO` leftover.

**Comment by psiinon** [comment]
Updated to support websocket replay :)

**Comment by psiinon** [comment]
@Pamplemousse those were all deliberately left in there .. and have now been emabled / implemented :D

**Comment by psiinon** [comment]
Hum, should probably disable Replay if the payload is not text, as we dont really support that - will implement that...

**Comment by psiinon** [comment]
Replay now disabled for non TEXT messages.
Looks like there no tutorial for websockets - my bad ;)
I'll add that in a future PR if thats ok? Maybe with break support ;)

**Comment by thc202** [comment]
The dialogue just allows to replay the message once, one has to reopen the dialogue to replay again.

**Comment by psiinon** [comment]
Good spot!
Simple option would be to close the dialog I guess...

**Comment by psiinon** [comment]
Changed to close the dialog after sending. This keeps it consistent with the HTTP one.
We can always change it in the future ...

**Pull request merged**